Mirror/Runtime/NetworkTransformBase.cs

```diff
@@ -0,0 +1,545 @@
+                float z = ScaleUShortToFloat(reader.ReadUInt16(), ushort.MinValue, ushort.MaxValue, 0, 360);
+                temp.rotation = Quaternion.Euler(x, y, z);
+            }
+            else if (compressRotation == Compression.Much)
+            {
+                // read 3 byte. scaling [0,255] to [0,360]
+                float x = ScaleByteToFloat(reader.ReadByte(), byte.MinValue, byte.MaxValue, 0, 360);
+                float y = ScaleByteToFloat(reader.ReadByte(), byte.MinValue, byte.MaxValue, 0, 360);
+                float z = ScaleByteToFloat(reader.ReadByte(), byte.MinValue, byte.MaxValue, 0, 360);
+                temp.rotation = Quaternion.Euler(x, y, z);
+            }
+            else if (compressRotation == Compression.Lots)
+            {
+                // read 2 byte, 5 bits per float
+                float[] xyz = UnpackUShortIntoThreeFloats(reader.ReadUInt16(), 0, 360);
+                temp.rotation = Quaternion.Euler(xyz[0], xyz[1], xyz[2]);
+            }
+
+            // timestamp
+            temp.timeStamp = Time.time;
+
+            // movement speed: based on how far it moved since last time
+            // has to be calculated before 'start' is overwritten
+            temp.movementSpeed = EstimateMovementSpeed(goal, temp, targetComponent.transform, GetNetworkSendInterval());
+
+            // reassign start wisely
+            // -> first ever data point? then make something up for previous one
+            //    so that we can start interpolation without waiting for next.
+            if (start == null)
+            {
+                start = new DataPoint{
+                    timeStamp=Time.time - GetNetworkSendInterval(),
+                    position=targetComponent.transform.position,
+                    rotation=targetComponent.transform.rotation,
+                    movementSpeed=temp.movementSpeed
+                };
+            }
+            // -> second or nth data point? then update previous, but:
+            //    we start at where ever we are right now, so that it's
+            //    perfectly smooth and we don't jump anywhere
+            //
+            //    example if we are at 'x':
+            //
+            //        A--x->B
+            //
+            //    and then receive a new point C:
+            //
+            //        A--x--B
+            //              |
+            //              |
+            //              C
+            //
+            //    then we don't want to just jump to B and start interpolation:
+            //
+            //              x
+            //              |
+            //              |
+            //              C
+            //
+            //    we stay at 'x' and interpolate from there to C:
+            //
+            //           x..B
+            //            \ .
+            //             \.
+            //              C
+            //
```

Comment:
Note that this can be good and bad.
For physics based movement if we move through a maze and then NetworkTransform tries to go through a wall, it might be bad.

It might be better to keep a list of data points and increase speed if we get too far behind (e.g. if list.size>2)

Reply:
That is how I do it in cubica,  a list of data point timestamped with NetworkTime.time